modules/ppcp-api-client/src/Repository/PartnerReferralsData.php

```diff
@@ -84,7 +84,7 @@ public function data(): array {
 					 */
 					'return_url'             => apply_filters(
 						'woocommerce_paypal_payments_partner_config_override_return_url',
-						admin_url( 'admin.php?page=wc-settings&tab=checkout&section=ppcp-gateway' )
+						add_query_arg( 'ppcp-return-url-nonce', wp_create_nonce( 'ppcp-return-url' ), admin_url( 'admin.php?page=wc-settings&tab=checkout&section=ppcp-gateway' ) )
```

Comment:
I think we were caching the onboarding URLs? The nonce may expire. Or if it's cached for one admin, then another admin will not be able to use it.